core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStandaloneTrainScheduleParser.java

```diff
@@ -67,24 +72,66 @@ private static MarecoAllowance parseAllowance(
                     new EnvelopeSimContext(rollingStock, envelopePath, timeStep),
                     rjsConstruction.beginPosition, rjsConstruction.endPosition,
                     getPositiveDoubleOrDefault(rjsConstruction.capacitySpeedLimit, 30 / 3.6),
-                    parseAllowanceValue(rjsConstruction.value));
+                    List.of(
+                            new AllowanceRange(
+                                    rjsConstruction.beginPosition,
+                                    rjsConstruction.endPosition,
+                                    parseAllowanceValue(rjsConstruction.value)
+                            )
+                    )
+            );
         }
         if (rjsAllowance.getClass() == RJSAllowance.Mareco.class) {
             var rjsMareco = (RJSAllowance.Mareco) rjsAllowance;
-            if (rjsMareco.ranges != null && rjsMareco.ranges.length > 0)
-                throw new InvalidSchedule("mareco value ranges aren't yet implemented");
             if (rjsMareco.defaultValue == null)
                 throw new InvalidSchedule("missing mareco default_value");
             return new MarecoAllowance(
                     new EnvelopeSimContext(rollingStock, envelopePath, timeStep),
                     0, envelopePath.getLength(),
                     getPositiveDoubleOrDefault(rjsMareco.capacitySpeedLimit, 30 / 3.6),
-                    parseAllowanceValue(rjsMareco.defaultValue));
+                    parseAllowanceRanges(envelopePath, rjsMareco.defaultValue, rjsMareco.ranges)
+            );
         }
 
         throw new RuntimeException("unknown allowance type");
     }
 
+    private static List<AllowanceRange> parseAllowanceRanges(
+            EnvelopePath envelopePath,
+            RJSAllowanceValue defaultValue,
+            RJSAllowanceRange[] ranges
+    ) throws InvalidSchedule {
+        // if no ranges have been defined, just return the default value
+        if (ranges == null || ranges.length == 0) {
+            return List.of(new AllowanceRange(0, envelopePath.getLength(), parseAllowanceValue(defaultValue)));
```

Comment:
can you call parseAllowanceValue(defaultValue) only once, and store the result in a local variable?